app/assets/javascripts/modules/all-content-finder.js

```diff
@@ -0,0 +1,53 @@
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {};
+
+(function (Modules) {
+  'use strict'
+
+  // Enhances the all content finder with dynamic behaviour and analytics tracking
+  class AllContentFinder {
+    constructor (module) {
+      this.module = module
+
+      this.$form = module.querySelector('.js-all-content-finder-form')
+      this.$taxonomySelect = module.querySelector('.js-all-content-finder-taxonomy-select')
+    }
+
+    init () {
+      this.setupTaxonomySelect()
+
+      if (this.userHasConsentedToAnalytics()) {
+        this.setupAnalyticsTracking()
+      } else {
+        // Allow tracking of events as soon as user consents, not just at next page reload
+        window.addEventListener('cookie-consent', this.setupAnalyticsTracking.bind(this))
```

Comment:
```suggestion
        window.addEventListener('cookie-consent', () => { this.setupAnalyticsTracking() })
```

Seems a bit more idiomatic ES6 to use an arrow function here.